lib/Jsrt/JsrtDiag.cpp

```diff
@@ -422,12 +422,42 @@ CHAKRA_API JsDiagSetStepType(
         }
         else if (stepType == JsDiagStepTypeStepOut)
         {
+#if TD_VSCODE_HIJACK_OUT_FOR_REVERSE_CONTINUE
+            ThreadContext* threadContext = runtime->GetThreadContext();
+
+            TTD::TTDebuggerSourceLocation bpLocation;
+            threadContext->TTDLog->GetPreviousTimeAndPositionForDebugger(bpLocation);
+            threadContext->TTDLog->SetPendingTTDBPInfo(bpLocation);
+
+            threadContext->TTDLog->LoadBPListForContextRecreate();
+
+            jsrtDebugManager->SetResumeType(BREAKRESUMEACTION_CONTINUE);
+#else
             jsrtDebugManager->SetResumeType(BREAKRESUMEACTION_STEP_OUT);
+#endif
         }
         else if (stepType == JsDiagStepTypeStepOver)
         {
             jsrtDebugManager->SetResumeType(BREAKRESUMEACTION_STEP_OVER);
         }
+        else if (stepType == JsDiagStepTypeStepBack)
+        {
+#if ENABLE_TTD
+            ThreadContext* threadContext = runtime->GetThreadContext();
+
+            TTD::TTDebuggerSourceLocation bpLocation;
+            threadContext->TTDLog->GetPreviousTimeAndPositionForDebugger(bpLocation);
+            threadContext->TTDLog->SetPendingTTDBPInfo(bpLocation);
+
+            threadContext->TTDLog->LoadBPListForContextRecreate();
+
+            //don't worry about BP suppression because we are just going to throw after we return
+
+            jsrtDebugManager->SetResumeType(BREAKRESUMEACTION_CONTINUE);
+#else
+            return JsErrorDiagUnableToPerformAction;
+#endif
```

Comment:
JsDiagStepTypeStepBack should only be allowed in replay mode of ttd and in all cases this should return JsErrorInvalidArgument